index.js

```diff
@@ -49,17 +49,24 @@ module.exports = (config) => {
         });
     }
 
-    /**
+     /**
      * Checks that the provided user is allowed to perform the action described by the request
-     *
-     * @param {object} req - The request.  Compatible with expressjs request.
-     *                       Required properties: method, path
-     * @param {string} username - The subject (or username) of the requester.
+     * This request is decoupled from the request path, allowing for an alternative description of resources
+     * 
+     * @param {object} abacRequest - The Attributes Based Access Control request.  
+     *                       Required properties: subject, resource, action 
      * @returns {promise} - A promise to authorize the user.
      *                      Resolves with the user and resource attributes.
      *                      Rejected if not authorized, or an error occurs.
      */
-    acs_utils.isAuthorized = (req, username) => {
+    acs_utils.isAuthorizedFor = (abacRequest) => {
+        if (abacRequest === null) {
+            throw new Error("Parameter: 'abacRequest' may not be null.");
```

Comment:
These errors should not be thrown here, but be inside the promise and reject it.  Otherwise the app may crash if the error is not handled.
e.g.
```
return new Promise((resolve, reject) => {
  if (!abacRequest) {
    return reject('Parameter: \'abacRequest\' may not be null.');
  }
  // Ensure we have a valid token to talk to ACS
  acs_utils._getToken().then((token) => {
    ...
```